src/lib/es2015.proxy.d.ts

```diff
@@ -1,21 +1,23 @@
-interface ProxyHandler<T extends object> {
+interface ProxyHandler<T extends object, U extends object> {
     apply?(target: T, thisArg: any, argArray: any[]): any;
     construct?(target: T, argArray: any[], newTarget: Function): object;
     defineProperty?(target: T, p: string | symbol, attributes: PropertyDescriptor): boolean;
     deleteProperty?(target: T, p: string | symbol): boolean;
-    get?(target: T, p: string | symbol, receiver: any): any;
+    get?<K extends keyof U>(target: T, p: K, receiver: U): U[K];
     getOwnPropertyDescriptor?(target: T, p: string | symbol): PropertyDescriptor | undefined;
     getPrototypeOf?(target: T): object | null;
     has?(target: T, p: string | symbol): boolean;
     isExtensible?(target: T): boolean;
     ownKeys?(target: T): ArrayLike<string | symbol>;
     preventExtensions?(target: T): boolean;
-    set?(target: T, p: string | symbol, value: any, receiver: any): boolean;
+    set?<K extends keyof U>(target: T, p: K, value: U[K], receiver: U): boolean;
     setPrototypeOf?(target: T, v: object | null): boolean;
 }
 
 interface ProxyConstructor {
-    revocable<T extends object>(target: T, handler: ProxyHandler<T>): { proxy: T; revoke: () => void; };
-    new <T extends object>(target: T, handler: ProxyHandler<T>): T;
+    revocable<T extends object>(target: T, handler: ProxyHandler<T, any>): { proxy: T; revoke: () => void; };
+    revocable<T extends object, U extends object>(target: T, handler: ProxyHandler<T, U>): { proxy: U; revoke: () => void; };
+    new <T extends object>(target: T, handler: ProxyHandler<T, any>): T;

```

Comment:
I can't immediately see how this ordering of overloads would work. Could you add tests where TypeScript choses one vs the other overload signatures?